mdm/mw/udidauth.go

```diff
@@ -0,0 +1,86 @@
+package mw
+
+import (
+	"bytes"
+	"context"
+	"crypto/x509"
+	"fmt"
+
+	"github.com/pkg/errors"
+
+	"github.com/micromdm/micromdm/mdm"
+	devicebuiltin "github.com/micromdm/micromdm/platform/device/builtin"
+)
+
+func UDIDAuthMiddleware(store *devicebuiltin.DB) mdm.Middleware {
+	return func(next mdm.Service) mdm.Service {
+		return &udidAuthMiddleware{
+			store: store,
+			next:  next,
+		}
+	}
+}
+
+type udidAuthMiddleware struct {
+	store *devicebuiltin.DB
```

Comment:
import or define an interface, but don't import the concrete implementation. Trying to phase these types of imports out to allow for a Postgres Implementation :)

This middleware only uses the `DeviceByUDID` and `Save` methods. IMO the best approach would be to define a local interface in this package with just those two methods.